python/lsst/meas/astrom/fitSipDistortion.py

```diff
@@ -0,0 +1,420 @@
+        exposure : :cpp:class:`lsst::afw::image::Exposure`
+            An Exposure or other displayable image on which matches can be
+            overplotted.  Ignored (and may be None) if display-based debugging
+            is not enabled via lsstDebug.
+
+        Returns
+        -------
+
+        An lsst.pipe.base.Struct with the following fields:
+
+        wcs : :cpp:class:`lsst::afw::image::TanWcs`
+            The best-fit WCS.
+        scatterOnSky : :cpp:class:`lsst::afw::geom::Angle`
+            The median on-sky separation between reference objects and
+            sources in "matches", as an lsst.afw.geom.Angle
+        """
+        import lsstDebug
+        display = lsstDebug.Info(__name__).display
+
+        if bbox is None:
+            bbox = lsst.afw.geom.Box2I()
+            for match in matches:
+                bbox.include(lsst.afw.geom.Point2I(match.second.getCentroid()))
+
+        wcs = self.makeInitialWcs(matches, initWcs)
+        cdMatrix = lsst.afw.geom.LinearTransform(wcs.getCDMatrix())
+
+        # Fit the "reverse" mapping from intermediate world coordinates to
+        # pixels, rejecting outliers. Fitting in this direction first makes it
+        # easier to handle the case where we have uncertainty on source
+        # positions but not reference positions.  That's the case we have
+        # right now for purely bookeeeping reasons, and it may be the case we
+        # have in the future when we us Gaia as the reference catalog.
+        revFitter = ScaledPolynomialTransformFitter.fromMatches(self.config.order, matches, wcs,
+                                                                self.config.refUncertainty)
+        revFitter.fit()
+        for nIter in range(self.config.numRejIter):
+            revFitter.updateModel()
+            intrinsicScatter = revFitter.updateIntrinsicScatter()
+            clippedSigma, nRejected = revFitter.rejectOutliers(self.outlierRejectionCtrl)
+            self.log.debug(
+                "Iteration {0}: intrinsic scatter is {1:4.3f} pixels, "
+                "rejected {2} outliers at {3:3.2f} sigma.".format(
+                    nIter+1, intrinsicScatter, nRejected, clippedSigma
+                )
+            )
+            if display or True:
+                self.display(revFitter, exposure=exposure, bbox=bbox)
+            revFitter.fit()
+        revScaledPoly = revFitter.getTransform()
+        # Convert the generic ScaledPolynomialTransform result to SIP form
+        # with given CRPIX and CD (this is an exact conversion, up to
+        # floating-point round-off error)
+        sipReverse = SipReverseTransform.convert(revScaledPoly, wcs.getPixelOrigin(), cdMatrix)
+
+        # Fit the forward mapping to a grid of points created from the reverse
+        # transform. Because that grid needs to be defined in intermediate
+        # world coordinates, we first grow the bounding box and then apply the
+        # initial WCS's CRPIX and CD matrix.
```

Comment:
I'm growing because I'm worried the transform I use to generate the box is only approximate, and I'd rather get a too-big box than a too-small one.  I'll clarify.